internal/security/bootstrapper/command/setupacl/command.go

```diff
@@ -0,0 +1,364 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package setupacl
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/config"
+	"github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/helper"
+	"github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/interfaces"
+
+	"github.com/edgexfoundry/go-mod-secrets/v2/pkg"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+)
+
+const (
+	// the command name for setting up registry's ACL
+	CommandName string = "setupRegistryACL"
+
+	consulGetLeaderAPI         = "/v1/status/leader"
+	consulACLBootstrapAPI      = "/v1/acl/bootstrap"
+	consulConfigAccessVaultAPI = "/v1/consul/config/access"
+	consulLegacyACLModeError   = "The ACL system is currently in legacy mode"
+	defaultRetryTimeout        = 30 * time.Second
+	emptyLeader                = `""`
+	emptyToken                 = ""
+)
+
+type cmd struct {
+	waitGroup     *sync.WaitGroup
+	loggingClient logger.LoggingClient
+	client        internal.HttpCaller
+	configuration *config.ConfigurationStruct
+
+	// internal state
+	errs         chan error
+	retryTimeout time.Duration
+}
+
+// NewCommand creates a new cmd and parses through options if any
+func NewCommand(
+	_ context.Context,
+	_ *sync.WaitGroup,
+	lc logger.LoggingClient,
+	conf *config.ConfigurationStruct,
+	args []string) (interfaces.Command, error) {
+	cmd := cmd{
+		waitGroup:     &sync.WaitGroup{},
+		loggingClient: lc,
+		client:        pkg.NewRequester(lc).Insecure(),
+		configuration: conf,
+		retryTimeout:  defaultRetryTimeout,
+	}
+	var dummy string
+
+	flagSet := flag.NewFlagSet(CommandName, flag.ContinueOnError)
+	flagSet.StringVar(&dummy, "confdir", "", "") // handled by bootstrap; duplicated here to prevent arg parsing errors
+
+	err := flagSet.Parse(args)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse command: %s: %w", strings.Join(args, " "), err)
+	}
+
+	return &cmd, nil
+}
+
+// Execute implements Command and runs this command
+// command setupRegistryACL sets up the ACL system of the registry, Consul in this case, preparing for generating
+// Consul's agent tokens later on
+func (c *cmd) Execute() (statusCode int, err error) {
+	c.loggingClient.Infof("Security bootstrapper running %s", CommandName)
+
+	if err := c.waitForNonEmptyConsulLeader(); err != nil {
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("failed to wait for Consul leader: %v", err)
+	}
+
+	var bootstrapTokenInfo *BootStrapACLTokenInfo
+	bootstrapTokenInfo, err = c.callConsulACLBootstrapAPI()
```

Comment:
Variables should not have the type in the name, so go short with simple `token` or ` aclToken` for name.

Reply:
ok